parser/dirs.go

```diff
@@ -88,6 +89,17 @@ func parseDir(fset *token.FileSet, dir, relPath string, filter func(os.FileInfo)
 				return nil, nil, err
 			}
 
+			// Check if this file should be part of the build
+			matched, err := context.MatchFile(dir, d.Name())
+			if err != nil {
+				errors.Add(token.Position{Filename: filename}, err.Error())
+				continue
+			}
+
+			if !matched {
```

Comment:
Nit: I'd remove the newline between the err check and the following if statement since they're closely connected